modelbaker/ilitoppingmaker/ilidata.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+                              -------------------
+        begin                : 2022-07-17
+        git sha              : :%H$
+        copyright            : (C) 2022 by Dave Signer
+        email                : david at opengis ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+import mimetypes
+import os
+import uuid
+import xml.dom.minidom as minidom
+import xml.etree.cElementTree as ET
+
+from .ilitarget import IliTarget
+
+
+class IliData(object):
+    class DatasetMetadata(object):
```

Comment:
Class in class... on purpose?
Also would be nice to have a class doc/description

Reply:
Yes on purpose. First I had all the classes in a module, so it felt nice in the structure.
But now - having a module ilidata.py - I think it can be not-nested as well. Might be better practice.